### PR TITLE
feat: add debug-flag for enabling logs in core-run

### PR DESCRIPTION
**Pull request by kushthedude**
Signed-off-by: kushthedude <kushthedude@gmail.com>

### CODE REVIEW

**Comment by kushthedude** [comment]
Hi @wonderflow , Please take a look at this!

File: cmd/core/main.go [hunk 1]
```diff
@@ -117,7 +118,7 @@ func main() {
 	}
 
 	logger := zap.New(func(o *zap.Options) {
-		o.Development = true
+		o.Development = logDebug
```
Comment (ryanzhang-oss) [inline cmd/core/main.go:121]: so it was true before, do you want to turn it off?

Comment (kushthedude) [reply cmd/core/main.go:121]: Hi @ryanzhang-oss , this was needed to give an additional debug-log flag to the user to turn on/off the lots of debug logs.

This is in result to https://github.com/crossplane/oam-kubernetes-runtime/issues/284#issue-737555765 https://github.com/crossplane/oam-kubernetes-runtime/pull/323#issuecomment-802575866

**Review by wonderflow — APPROVED** [review]
LGTM

**Pull request merged**